### PR TITLE
add support for app portal token

### PR DESCRIPTION
**Pull request by Dotunj**

### CODE REVIEW

**Review by ogbanugot — APPROVED** [review]

File: server/middleware.go [hunk 1]
```diff
@@ -147,6 +148,81 @@ func requireApp(appRepo datastore.ApplicationRepository) func(next http.Handler)
...
+			}
+
+			if !authUser.Role.Type.Is(role) {
+				_ = render.Render(w, r, newErrorResponse("unauthorized role", http.StatusUnauthorized))
+				return
+			}
+
+			group := getGroupFromContext(r.Context())
+			for _, v := range authUser.Role.Groups {
+				if group.Name == v || group.UID == v {
```
Comment (Subomi) [inline server/middleware.go:200]: @Dotunj Since we already decided that applications should be unique is there any point in doing a group check again?

**Review by Subomi — APPROVED** [review]

File: server/security.go [hunk 2]
```diff
@@ -112,6 +113,100 @@ func (a *applicationHandler) CreateAPIKey(w http.ResponseWriter, r *http.Request
...
+	if err != nil {
+		log.WithError(err).Error("failed to create api key")
+		_ = render.Render(w, r, newErrorResponse("failed to create api key", http.StatusInternalServerError))
+		return
+	}
+
+	baseUrl := getBaseUrlFromContext(r.Context())
+
+	if util.IsStringEmpty(baseUrl) {
+		baseUrl = "<insert-base-url-in-config>"
```
Comment (Subomi) [inline server/security.go:196]: @Dotunj What is meant to be here?

File: server/security.go [hunk 3]
```diff
@@ -112,6 +113,100 @@ func (a *applicationHandler) CreateAPIKey(w http.ResponseWriter, r *http.Request
...
+		Type:           "app_portal",
+		Role:           role,
+		Hash:           encodedKey,
+		Salt:           salt,
+		CreatedAt:      primitive.NewDateTimeFromTime(time.Now()),
+		UpdatedAt:      primitive.NewDateTimeFromTime(time.Now()),
+		DocumentStatus: datastore.ActiveDocumentStatus,
+	}
+
+	if newApiKey.ExpiresAt != (time.Time{}) {
```
Comment (Subomi) [inline server/security.go:182]: @Dotunj Users are not meant to supply the expiration time of these keys. We should set it to a reasonable default - 30 minutes?

**Review by Subomi — CHANGES_REQUESTED** [review]
@Dotunj Drop some comments that require attention. :(

**Comment by Dotunj** [comment]
@Subomi addressed your comments

**Review by Subomi — APPROVED** [review]

**Pull request merged**